### PR TITLE
Adding CRD for adding users

### PR DESCRIPTION
**Pull request by roller1187**

### CODE REVIEW

File: config/crd/kustomization.yaml [hunk 1]
```diff
@@ -6,4 +6,5 @@ kind: Kustomization
 # It should be run by config/default
 resources:
 - bases/redhatgov.io_giteas.yaml
+- bases/redhatgov.io_giteasuser.yaml
```
Comment (andykrohg) [inline config/crd/kustomization.yaml]: Could you please change this to `giteausers` 🙂

**Review by jflowers — APPROVED** [review]
lgtm

**Review by jflowers — CHANGES_REQUESTED** [review]
actually, let's be consitent point to a secret and not store the password in the CR

File: config/crd/kustomization.yaml [hunk 2]
```diff
@@ -6,4 +6,5 @@ kind: Kustomization
 # It should be run by config/default
 resources:
 - bases/redhatgov.io_giteas.yaml
+- bases/redhatgov.io_giteauser.yaml
```
Comment (andykrohg) [inline config/crd/kustomization.yaml]: Can you change this to `giteausers.yaml`? 😄

**Pull request merged**